File: techila/examples/Matlab/cloudfor/ad_impersonate/run_impersonate.m
function run_impersonate()
% This function contains the cloudfor-loop, which will be used to distribute
% computations to the Techila environment.
%
% During the computational Project, Active Directory impersonation will be 
% used to run the Job under the End-User's own AD user account.
%
% Syntax:
%
% run_impersonate()

% Copyright 2015 Techila Technologies Ltd.

% Check which user account is used locally
[s,local_username]=dos('whoami');

% The keywords 'cloudfor' and 'cloudend' have been used to mark the
% beginning and end of the block of code that will be executed on Workers.
%
% AD impersonation has been enabled by setting the value of the 
% 'techila_ad_impersonate' Project parameter to true.
cloudfor x=1:1
    %peach('ProjectParameters',{{'techila_ad_impersonate','true'}})
    
    % Check which user account is used to run the computational Job.
    [s,worker_username]=dos('whoami');
cloudend

% Print the user account information
fprintf('Username on local computer: %s\n',local_username);
fprintf('Username on Worker computer: %s\n',worker_username);
end

File: techila/examples/R/cloudfor/ad_impersonate/run_impersonate.r
run_impersonate <- function() {
# This function contains the cloudfor-loop, which will be used to distribute
# computations to the Techila environment.
#
# During the computational Project, Active Directory impersonation will be 
# used to run the Job under the End-User's own AD user account.
#
# Syntax:
#
# source("run_impersonate.r")
# res <- run_impersonate()

# Copyright 2015 Techila Technologies Ltd.

  # Load the techila package
  library(techila)
  
  # Check which user account is used locally
  local_username <- system("whoami",intern=TRUE)
  
  worker_username <-  cloudfor (i=1:1, # Set the maximum number of iterations to one
                                .sdkroot="../../../..", # Location of the Techila SDK 'techila' directory
                                .ProjectParameters = list("techila_ad_impersonate" = "true") # Enable AD impersonation
  ) %t% {
    # Check which user account is used to run the computational Job.
    worker_username <- system("whoami",intern=TRUE)
  }
  
  # Print and return the results
  cat("Username on local computer:",local_username, "\n")
  cat("Username on Worker computer:",worker_username, "\n")
  list(local_username,worker_username)
}
